TheengsGateway/__init__.py

```diff
@@ -32,9 +32,11 @@
     "pass":"",
     "ble_scan_time":5,
     "ble_time_between_scans":5,
-    "publish_topic": "home/TheengsGateway/BTtoMQTT",
+    "publish_topic": "home/TheengsGatewayBTtoMQTT",
     "subscribe_topic": "home/TheengsGateway/commands",
-    "log_level": "WARNING"
+    "log_level": "WARNING",
+    "discovery": False,
+    "discovery_topic": "homeassistant/sensor/TheengsGateway" 
```

Comment:
There is an attribute in the discovery message that gives the gateway that published the message `"via_device":"OMG_GAT"`. This enables to identify from which gateway the publishing message was done.
Gateways are a passthrough for sensors, this structure preserves the unicity of the sensor topics.

Reply:
How about a setting for this? I mean, we could let the user decide

Reply:
if we can let it by default to:
`<discovery_prefix>/<component>/<object_id>/config` I'm ok

Reply:
Currently, for discovery I publish to `homeassistant/sensor/[unique_id]/[property]` with subtopics for config, attributes and state. Are you ok with this? Or should I only have *homeassistant/sensor/[unique_id]/[propery}/config` with the state and attributes published to corresponding subtopics from `homeassistant/OMG/BTtoMQTT/[unnique_id]/[property] much like OMG?

Reply:
I would say to stick per default to OMG structure which is:
`homeassistant/sensor/<object_id>-<property/config`
Example:
`homeassistant/sensor/882D34677766-lux/config`

I don't think you need to have a `state` subtopic as we are not controlling the BLE devices, [referring to this example](https://www.home-assistant.io/docs/mqtt/discovery/#sensors-with-multiple-values).

> BTW, for the Gateway I´ll work on controls, since I´m gonna add an allow-list, atleast for discovery.

You can take inspiration from [this API](https://docs.openmqttgateway.com/use/ble.html#setting-a-white-or-black-list)